diff --git a/Makefile b/Makefile
index d1cbc3a..22983a2 100644
--- a/Makefile
+++ b/Makefile
@@ -1,4 +1,4 @@
-PREFIX := /usr
+PREFIX := /opt/3rd-party/bundles/clearfraction/usr
 
 USE_SYSTEMD ?= 1
 USE_DBUS_ACTIVATION ?= 1
diff --git a/systemd/waydroid-container.service b/systemd/waydroid-container.service
index 4a86383..1c17e6a 100644
--- a/systemd/waydroid-container.service
+++ b/systemd/waydroid-container.service
@@ -4,7 +4,13 @@ Description=Waydroid Container
 [Service]
 UMask=0022
 BusName=id.waydro.Container
-ExecStart=/usr/bin/waydroid -w container start
+EnvironmentFile=-/tmp/waydroid.env
+ExecStartPre=sh -c 'env | grep -v "^PATH=" > /tmp/waydroid.env'
+ExecStartPre=sh -c 'echo PYTHONPATH=/opt/3rd-party/bundles/clearfraction/usr/lib/python$(echo /usr/lib/python* | tr -d "/usr/lib/python")/site-packages >> /tmp/waydroid.env'
+ExecStartPre=sh -c 'echo PATH=/usr/bin/haswell:/usr/bin:/usr/local/bin:/opt/3rd-party/bundles/clearfraction/bin:/opt/3rd-party/bundles/clearfraction/usr/bin:/opt/3rd-party/bundles/clearfraction/usr/local/bin:/opt/3rd-party/bin >> /tmp/waydroid.env'
+ExecStartPre=sh -c 'echo LD_LIBRARY_PATH=/usr/lib64:/opt/3rd-party/bundles/clearfraction/usr/lib64:/opt/3rd-party/bundles/clearfraction/usr/local/lib64 >> /tmp/waydroid.env'
+ExecStart=/opt/3rd-party/bundles/clearfraction/usr/bin/waydroid -w container start
+ExecStartPost=sh -c 'rm /tmp/waydroid.env'
 
 [Install]
 WantedBy=multi-user.target
diff --git a/tools/config/__init__.py b/tools/config/__init__.py
index 002562e..55a0c9f 100644
--- a/tools/config/__init__.py
+++ b/tools/config/__init__.py
@@ -36,7 +36,7 @@ defaults = {
     "vendor_datetime": "0",
     "preinstalled_images_paths": [
         "/etc/waydroid-extra/images",
-        "/usr/share/waydroid-extra/images",
+        "/opt/3rd-party/bundles/clearfraction/usr/share/waydroid-extra/images",
     ],
     "suspend_action": "freeze",
     "mount_overlays": "True",
@@ -72,7 +72,7 @@ if session_defaults["pulse_runtime_path"] == "None":
     session_defaults["pulse_runtime_path"] = session_defaults["xdg_runtime_dir"] + "/pulse"
 
 channels_defaults = {
-    "config_path": "/usr/share/waydroid-extra/channels.cfg",
+    "config_path": "/opt/3rd-party/bundles/clearfraction/usr/share/waydroid-extra/channels.cfg",
     "system_channel": "https://ota.waydro.id/system",
     "vendor_channel": "https://ota.waydro.id/vendor",
     "rom_type": "lineage",
